### ISSUE TITLE
Smoke tests failing in stage 2 with `VSTESTTASK2` error

### ISSUE THREAD

**Issue by mthalman** [dotnet/source-build#4003]
```
VSTESTTASK2 : error Message:  [/vmr/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj]
```

No other information exists. It's an error contained in the build output but not an actual test case failure. [Example build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2359219&view=logs&j=b915386b-fe84-5c50-207a-7441ae49c960&t=0e82c39b-dfd6-5dfb-430f-d39c8d3de7c2) (internal link)

It would seem this is introduced by new functionality from within vstest that gets compiled into the SDK and used in the stage 2 build.

**Comment by mthalman** [comment]
This may be a regression caused by https://github.com/microsoft/vstest/pull/4834. These tests were working last Friday and failing today after incorporating changes that included https://github.com/microsoft/vstest/pull/4834. @nohwnd - Can you take a look? 
Here's a binlog: [BuildTests_0122193259.zip](https://github.com/dotnet/source-build/files/14014935/BuildTests_0122193259.zip). We need to get this resolved soon for .NET 9 Preview 1.

**Comment by mthalman** [comment]
I ran a [test build](https://dev.azure.com/dnceng/internal/_git/dotnet-dotnet/commit/b404e7f7756a64d9d6e5c73e18ed4353fa239619/) (internal link) that reverted the change from https://github.com/microsoft/vstest/pull/4834 but that didn't help at all.

**Comment by nohwnd** [comment]
This is more likely linked to this change: https://github.com/dotnet/sdk/pull/38098/files which changes how failed tests are reported to take advantage of msbuild terminal logger which is now the default experience for `dotnet build`. 

I am trying to repro to see if the smoke tests failed and VSTEST fails to report the details of the failure. Or if VSTEST is failing and the change needs to be reverted.

**Comment by nohwnd** [comment]
Running another test build with opt-out from the new feature https://dev.azure.com/dnceng/internal/_build/results?buildId=2360131&view=results 

In the meantime I am trying to repro locally.

**Comment by nohwnd** [comment]
I can't repro fully. 

I can run the tests but I always get the full output of the error. And I can't run tests without failures locally because I don't have all the artifacts. I always see: `Tarball path '' specified in  does not exist.` ( [from here](https://github.com/dotnet/installer/blob/main/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/DotNetHelper.cs#L38) ) I downloaded what I could but I still cannot get it to work locally. 

I can see in binary logger the same error output as in build, but it is followed by the actual error message. 

In your run I cannot see any failures in trx, so I am not sure where the errors are coming from and if they are real.

I've edited the target to actually propagate the opt-out flag and started another build here (from a new branch, I want to see how the other one will finish) https://dev.azure.com/dnceng/internal/_build/results?buildId=2360247&view=results
